core/modules.rs

```diff
@@ -267,7 +273,7 @@ impl<L: Loader> Future for RecursiveLoad<L> {
       let mut resolve_cb =
         |specifier: &str, referrer_id: deno_mod| -> deno_mod {
           let referrer = modules.get_name(referrer_id).unwrap();
-          match L::resolve(specifier, &referrer) {
+          match L::resolve(&self.loader.unwrap(), specifier, &referrer, true) {
```

Comment:
Can't get this bit past borrow checker 😭 `self.loader` is consumed when acquiring `isolate` and `modules`.

Reply:
I think we need to refactor Loader to not own the Isolate. This should be done as a separate PR. I will take a stab at it....

Reply:
Awesome, looking forward to it